macros/src/main/scala/org/zalando/grafter/macros/ReaderMacros.scala

```diff
@@ -1,91 +1,90 @@
 package org.zalando.grafter.macros
 
-import scala.reflect.macros.whitebox
+import scala.meta._
 
 object ReaderMacros {
 
   /** get the annotated class and, if available, companion object */
-  def annotationInputs(c: whitebox.Context)(name: String)(annottees: Seq[c.Expr[Any]]): (c.universe.Tree, Option[c.universe.Tree]) =
-    annottees.toList match {
-      case classDecl :: companion :: _ =>
-      (classDecl.tree, Option(companion.tree))
+  def annotatedClass(name: String)(annotated: Tree): (Defn.Class, Option[Defn.Object]) =
+    annotated match {
+      case block: Term.Block =>
+        block.stats.toList match {
+          case (classDef: Defn.Class) :: (companionDef: Defn.Object) :: _ =>
```

Comment:
Amazingly yes. This is something that's a bit hard on people who want to design the macro system. You have an annotation on a class and you want to grab the corresponding companion object if it exists. But it might not be declared next to the class, it could be before or separated by something else. Magically the macro must give you both.